server/src/test/java/org/elasticsearch/health/node/HealthInfoTests.java

```diff
@@ -54,7 +63,7 @@ public static HealthInfo mutateHealthInfo(HealthInfo originalHealthInfo) {
                 HealthInfoTests::randomRepoHealthInfo
             );
         }
-        return new HealthInfo(diskHealth, dslHealth, repoHealth);
+        return new HealthInfo(diskHealth, dslHealth, repoHealth, mutateFileSettingsHealthInfo(fsHealth));
```

Comment:
Can you move the mutation to the `switch` above? That's more in-line with other tetst of this type and avoids confusion.